test/e2e/all_in_one_test.go

```diff
@@ -119,28 +84,50 @@ func allInOneWithIngressTest(t *testing.T, f *framework.Framework, ctx *framewor
 		return err
 	}
 
-	ingress, err := WaitForIngress(t, f.KubeClient, namespace, "my-jaeger-with-ingress-query", retryInterval, timeout)
-	if err != nil {
-		return err
-	}
+	var url string
+	var httpClient http.Client
+	if isOpenShift(t) {
+		err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, name, 1, retryInterval, 3 * timeout)
+		if err != nil {
+			t.Errorf("Error waiting for deployment of %s: %v\n", name, err)
+			return err
+		}
 
-	if len(ingress.Status.LoadBalancer.Ingress) != 1 {
-		return fmt.Errorf("Wrong number of ingresses. Expected 1, was %v", len(ingress.Status.LoadBalancer.Ingress))
+		routeUrl, err := findRouteUrl(t, f, name)
+		if err != nil {
+			return err
+		}
+
+		url = fmt.Sprintf("https://%s/api/services", routeUrl)
+		transport := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+		httpClient= http.Client{Timeout: 30 * time.Second, Transport: transport}
+	} else {
+		ingress, err := WaitForIngress(t, f.KubeClient, namespace, "my-jaeger-with-ingress-query", retryInterval, timeout)
```

Comment:
Wondering why the names are different?
@jpkrohling I think this is because the route is named with `-query` whereas the ingress isn't. Do we actually need this distinction or would it be better to be consistent? (either way - i.e. add `-query` to ingress or remove it from route)?

Reply:
AFAIR, we don't access ingresses by its name, because it might need some DNS changes. So, we get the ingress IP and use it instead. Routes create a hostname based on the backing service name (query in our case) that could probably be reached by the client if it uses something like `nip.io`.

Reply:
Sounds reasonable - was mistaken anyway, route and ingress are both named with `-query` suffix. So ok as is - only possible change could be to move the `WaitForDeployment` above the if/else as independent of whether using openshift or k8s?

Reply:
I could move the WaitForDeployment, but in the Kubernetes case WaitForIngress serves the same purpose.  Let me know what you prefer, I can do it either way.

Reply:
Might be better to move, as not sure whether the ingress being available guarantees that the deployment is available - so moving might avoid issues later with timing.